A job failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/10067057346/job/27829790907) [(plain)](https://github.com/rust-lang-ci/rust/commit/371dee1766ca104d7a3504111ac3f73048b8cc3e/checks/27829790907/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
2
Image input checksum f5a89363500623842dde699873e3443e077792d5c1d7f9eacb25433aaf7830dd38c2d021b3a39dbc90b7037b2d09847fb23911ef436200cd54356efabe20474b
##[group]Building docker image for aarch64-gnu
Docker version 26.1.3, build b72abbb
Error response from daemon: Get "https://ghcr.io/v2/": Get "https://ghcr.io/token?account=rust-lang-ci&client_id=docker&offline_token=true&service=ghcr.io": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers) (Client.Timeout exceeded while awaiting headers)
##[error]Process completed with exit code 1.
```

</details>

--- a/compiler/rustc_trait_selection/src/traits/select/mod.rs
+++ b/compiler/rustc_trait_selection/src/traits/select/mod.rs
@@ -1498,7 +1498,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             return false;
         }
 
-        // Avoid using the master cache during coherence and just rely
+        // Avoid using the global cache during coherence and just rely
         // on the local cache. This effectively disables caching
         // during coherence. It is really just a simplification to
         // avoid us having to fear that coherence results "pollute"
@@ -1506,6 +1506,12 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         // it's not worth going to more trouble to increase the
         // hit-rate, I don't think.
         if self.is_intercrate() {
+            return false;
+        }
+
+        // Avoid using the global cache when we're defining opaque types
+        // as their hidden type may impact the result of candidate selection.
+        if !self.infcx.defining_opaque_types().is_empty() {
             return false;
         }
 

--- a/tests/ui/auto-traits/opaque_type_candidate_selection.rs
+++ b/tests/ui/auto-traits/opaque_type_candidate_selection.rs
@@ -1,4 +1,7 @@
-//@ known-bug: #119272
+//! used to ICE: #119272
+
+//@ check-pass
+
 #![feature(type_alias_impl_trait)]
 mod defining_scope {
     use super::*;

